src/main/java/net/dv8tion/jda/internal/entities/TextChannelImpl.java

```diff
@@ -456,16 +456,27 @@ public MessageAction sendFile(@Nonnull InputStream data, @Nonnull String fileNam
     public RestActionImpl<Void> removeReactionById(@Nonnull String messageId, @Nonnull String unicode, @Nonnull User user)
     {
         Checks.isSnowflake(messageId, "Message ID");
-        Checks.noWhitespace(unicode, "Unicode emoji");
+        Checks.notNull(unicode, "Provided Unicode");
+        unicode = unicode.trim();
+        Checks.notEmpty(unicode, "Provided Unicode");
         Checks.notNull(user, "User");
+
         if (!getJDA().getSelfUser().equals(user))
             checkPermission(Permission.MESSAGE_MANAGE);
-        final String code = EncodingUtil.encodeUTF8(unicode);
-        Route.CompiledRoute route;
+
+        String encoded;
+        if (unicode.startsWith("U+") || unicode.startsWith("u+"))
+            encoded = EncodingUtil.encodeCodepointsUTF8(unicode);
+        else
+            encoded = EncodingUtil.encodeUTF8(unicode);
```

Comment:
We should move this if/else into `EncodingUtil` as something like `encodeReaction(String)`